Mod2/Lessons/Week5/DatabaseMigrations.md

```diff
@@ -0,0 +1,135 @@
+
+> With you group: Brainstorm business decisions that would cause a developer to modify the database table. What exactly would need to change in the database?
+
+Here are some sentence starters to help you get started!
+
+**One business decision that probably to a database change was when __________ decided to __________. The developers probably needed to update the database by ______________________.**
+
+### Implementing DB Changes
+
+Let's work together to make a DB Change in our plant app. We now want to keep track of when each plant was last watered so that we can remind our users when to water their plants. Let's call the column `last_watered` and make it type DateTime.
+
+I'm going to pull up the documentation from the preparation to use as a reference. https://learn.microsoft.com/en-us/ef/core/managing-schemas/migrations/?tabs=dotnet-core-cli
+
+>"At a high level, migrations function in the following way:
+>When a data model change is introduced, the developer uses EF Core tools to add a corresponding migration describing the updates necessary to keep the database schema in sync. EF Core compares the current model against a snapshot of the old model to determine the differences, and generates migration source files; the files can be tracked in your project's source control like any other source file." - The Migration Documentation
+
+#### Existing Records After Adding New Column
+
+Take a look at what happened to the existing records when we added a new column. There needs to be some value for the column and the default is -infinity. It's a bit of a confusing value, but -infinity means it is earlier than all other timestamps.
+
+<p align='center'>
+  <img src='../../Images/Week5/new_column_infinity.png'>
+</p>
+
+#### Deleting a Column
+
+With your partner, to try figure out how to delete a column. Feel free to try things, search in the documentation, or use google to try and figure it out!
+
+<!-- Drop down here -->
+The data that existed in that column will be lost, but deleting the column only requires removing the field from the model, making a migration, and updating the database to apply that migration.
+
+While this guide is for a SQLite database (another type of SQL database), the process is the same when using postgreSQL. https://www.learnentityframeworkcore5.com/whats-new-in-ef-core-5/drop-column-from-sqlite-database
+
+#### Up and Down Sections of Migrations
+
+You may have noticed that each migration file has an `Up` function and a `Down` function.
+
+> With you partner: What do you think these two functions are for? You can look at the screenshot below or the migrations in your plant tracker application. Google may also be helpful here!
+
+<p align='center'>
+  <img src='../../Images/Week5/migration_up_and_down.png'>
+</p>
+
+❓Do we want to store our migration files in Github?
```

Comment:
Do we want students to answer this question? or is this a discussion starter?

Reply:
I'm thinking of these ❓ questions as deeper more thought-provoking questions to ask the students and give them time to think on, but thinking of leaving it to the instructor of how to ask them. Maybe time to reflect and then a chat waterfall? Maybe think-pair-share?